src/object_schema.cpp

```diff
@@ -107,11 +108,36 @@ Property *ObjectSchema::property_for_name(StringData name) {
     return nullptr;
 }
 
-const Property *ObjectSchema::property_for_name(StringData name) const {
+Property *ObjectSchema::property_for_public_name(StringData public_name)
+{
+    // If no `public_name` is defined, the internal `name` is also considered the public name.
+    for (auto& prop : persisted_properties) {
+        if ((prop.public_name.empty() ? StringData(prop.name) :  StringData(prop.public_name)) == public_name)
+            return &prop;
+    }
+
+    // Computed properties are not persisted, so creating a public name for such properties
+    // are a bit pointless since the internal name is already the "public name", but since
+    // this distinction isn't visible in the Property struct we allow it anyway.
+    for (auto& prop : computed_properties) {
+        if ((prop.public_name.empty() ? StringData(prop.name) :  StringData(prop.public_name)) == public_name)
```

Comment:
For clarity, it might be worth splitting them into two separate statements. I mean, having a `?` in an `if` (branching in the condition for branching) :smile: